bpf/kprobe_pwru.c

```diff
@@ -216,12 +217,12 @@ filter_l3_and_l4(struct sk_buff *skb, struct config *cfg) {
 
 		if (l4_proto == IPPROTO_TCP) {
 			struct tcphdr *tcp = (struct tcphdr *) (skb_head + l4_off);
-			sport = BPF_CORE_READ(tcp, source);
-			dport = BPF_CORE_READ(tcp, dest);
+			sport = bpf_ntohs(BPF_CORE_READ(tcp, source));
```

Comment:
I would keep the endian conversions in the user space as it is to save a few CPU cycles. Otherwise, the conversions would happen for each packet.

Reply:
You are right. Now I have noticed that the conversion is unnecessary when we do not output tuples.